openfeature/evaluator.go

```diff
@@ -0,0 +1,445 @@
+	}
+
+	// Check if attribute value matches any item in the list
+	for _, item := range conditionList {
+		if matchesValue(attributeValue, item) {
+			return true
+		}
+	}
+	return false
+}
+
+// matchesValue checks if an attribute value equals a string value with type coercion.
+func matchesValue(attributeValue any, strValue string) bool {
+	switch av := attributeValue.(type) {
+	case string:
+		return av == strValue
+	case int, int8, int16, int32, int64:
+		i64, err1 := toInt64(av)
+		parsed, err2 := strconv.ParseInt(strValue, 10, 64)
+		return err1 == nil && err2 == nil && i64 == parsed
+	case uint, uint8, uint16, uint32, uint64:
+		u64, err1 := toUint64(av)
+		parsed, err2 := strconv.ParseUint(strValue, 10, 64)
+		return err1 == nil && err2 == nil && u64 == parsed
+	case float32, float64:
+		f64, ok := internal.ToFloat64(av)
+		parsed, err2 := strconv.ParseFloat(strValue, 64)
+		return ok && err2 == nil && f64 == parsed
+	case bool:
+		parsed, err := strconv.ParseBool(strValue)
+		return err == nil && av == parsed
+	default:
+		return fmt.Sprintf("%v", av) == strValue
+	}
+}
+
+// evaluateNumericCondition evaluates numeric comparison operators.
+func evaluateNumericCondition(attributeValue any, conditionValue any, operator conditionOperator) bool {
+	attrNum, ok := internal.ToFloat64(attributeValue)
+	if !ok {
+		return false
+	}
+
+	condNum, ok := internal.ToFloat64(conditionValue)
+	if !ok {
+		return false
+	}
+
+	switch operator {
+	case operatorGT:
+		return attrNum > condNum
+	case operatorGTE:
+		return attrNum >= condNum
+	case operatorLT:
+		return attrNum < condNum
+	case operatorLTE:
+		return attrNum <= condNum
+	default:
+		return false
+	}
+}
+
+// evaluateSplit determines if a split matches by evaluating all its shards.
+func evaluateSplit(split *split, context map[string]any) bool {
+	// All shards must match (AND logic)
+	for _, shard := range split.Shards {
+		if !evaluateShard(shard, context) {
+			return false
+		}
+	}
+	return true
+}
+
+// evaluateShard evaluates a shard using consistent hashing.
+func evaluateShard(shard *shard, context map[string]any) bool {
+	// Get targeting key from context
+	targetingKey := getTargetingKey(context)
+	if targetingKey == "" {
+		return false
+	}
+
+	// Compute shard index using MD5 hash (matching Eppo's implementation)
+	shardIndex := computeShardIndex(shard.Salt, targetingKey, shard.TotalShards)
+
+	// Check if shard index falls within any of the ranges
+	for _, shardRange := range shard.Ranges {
+		if shardIndex >= shardRange.Start && shardIndex < shardRange.End {
+			return true
+		}
+	}
+	return false
+}
+
+// getTargetingKey extracts the targeting key from the context.
+// It checks for "targetingKey" first, then falls back to other common keys.
+func getTargetingKey(context map[string]any) string {
+	// Try standard OpenFeature targeting key
+	if key, ok := context["targetingKey"]; ok {
+		if str, ok := key.(string); ok {
+			return str
+		}
+	}
+
+	// Try common alternatives
```

Comment:
I like how you are thinking about the UX but this risks being "too magical" and could accidentally cause randomization against an undesired key. just `targetingKey` is sufficient.